src/features/ExtensionCommands.ts

```diff
@@ -408,6 +416,74 @@ export class ExtensionCommandsFeature implements IFeature {
         return promise;
     }
 
+    private saveFile(filePath: string): Thenable<EditorOperationResponse> {
+
+        var promise: Thenable<EditorOperationResponse>;
+
+        var normalizedFilePath = this.normalizeFilePath(filePath);
+
+        // since Windows is case-insensitive, we need to normalize it differently
+        var canFind = vscode.workspace.textDocuments.find(doc => {
```

Comment:
Probably also good to make this its own function to avoid duplication

Reply:
Yep. Looks much cleaner now. Thanks!